### ISSUE TITLE
DeepIV negative loss function

### ISSUE THREAD

**Issue by Yannnnc** [microsoft/EconML#628]
Hi, 
I tried to use the deepiv function with the microsoft tutorial and find the loss value is negative at first and then turns to positive. Here are also two reminders that "output model_344 and model_350 missing from loss dictionary". 

How should I fixed this problem? Your help is much appreciated! 
Here's my code, I have tried the model with my original data and normalized data:

treatment_model = keras.Sequential([keras.layers.Dense(128, activation='relu', input_shape=(5,)),
                                    keras.layers.Dropout(0.0001),
                                    keras.layers.Dense(64, activation='relu'),
                                    keras.layers.Dropout(0.0001),
                                    keras.layers.Dense(32, activation='relu'),
                                    keras.layers.Dropout(0.0001)])

response_model = keras.Sequential([keras.layers.Dense(128, activation='relu', input_shape=(5,)),
                                   keras.layers.Dropout(0.0001),
                                   keras.layers.Dense(64, activation='relu'),
                                   keras.layers.Dropout(0.0001),
                                   keras.layers.Dense(32, activation='relu'),
                                   keras.layers.Dropout(0.0001),
                                   keras.layers.Dense(1)])

deepIvEst = DeepIV(n_components = 10, # number of gaussians in our mixture density network
                   m = lambda z, x : treatment_model(keras.layers.concatenate([z,x])), # treatment model
                   h = lambda t, x : response_model(keras.layers.concatenate([t,x])),  # response model
                   n_samples = 1, # number of samples to use to estimate the response
                   use_upper_bound_loss = False, # whether to use an approximation to the true loss
                   n_gradient_samples = 1, # number of samples to use in second estimate of the response (to make loss estimate unbiased)
                   optimizer='Adagrad', # Keras optimizer to use for training - see https://keras.io/optimizers/ 
                   first_stage_options = keras_fit_options, # options for training treatment model
                   second_stage_options = keras_fit_options) # options for training response model

deepIvEst.fit(Y=y,T=t,X=x,Z=z)

![截屏2022-05-27 下午10 09 23](https://user-images.githubusercontent.com/106391157/170716085-09376a5e-b9ac-4e26-8dd7-c6c5ae4e0f2c.png)
